liquibase-service/src/main/resources/db/rdb_modern/routines/018-sp_d_lab_test_postprocessing-001.sql

```diff
@@ -3085,22 +3086,26 @@ BEGIN
         SET @PROC_STEP_NO = @PROC_STEP_NO + 1;
         SET @PROC_STEP_NAME = 'Remove LAB_TEST Results';
 
-        /* Removed records associated to Deleted Orders. */
-        WITH removed_obs as (
-            select LAB_TEST_UID,
-                   PARENT_TEST_PNTR,
-                   ROOT_ORDERED_TEST_PNTR from dbo.LAB_TEST
-            where ROOT_ORDERED_TEST_PNTR = 1
-
-            UNION ALL
-
-            select lt.LAB_TEST_UID,
-                   lt.PARENT_TEST_PNTR,
-                   lt.ROOT_ORDERED_TEST_PNTR
-            from dbo.LAB_TEST lt
-                     inner join removed_obs o on lt.PARENT_TEST_PNTR = o.LAB_TEST_UID
-        )
-        DELETE FROM dbo.LAB_TEST WHERE LAB_TEST_UID IN (SELECT LAB_TEST_UID FROM removed_obs);
+        /* Remove records associated to Deleted Results. */
+        IF EXISTS (SELECT 1 FROM dbo.LAB_TEST WHERE ROOT_ORDERED_TEST_PNTR = 1)
+            BEGIN
+                WITH removed_obs as (
+                    select LAB_TEST_UID,
+                           PARENT_TEST_PNTR,
+                           ROOT_ORDERED_TEST_PNTR from dbo.LAB_TEST
+                    where ROOT_ORDERED_TEST_PNTR = 1
+
+                    UNION ALL
+
+                    select lt.LAB_TEST_UID,
+                           lt.PARENT_TEST_PNTR,
+                           lt.ROOT_ORDERED_TEST_PNTR
+                    from dbo.LAB_TEST lt
+                             inner join removed_obs o on lt.PARENT_TEST_PNTR = o.LAB_TEST_UID
+                )
+                DELETE FROM dbo.LAB_TEST WHERE LAB_TEST_UID IN (SELECT LAB_TEST_UID FROM removed_obs);
+            END
+
```

Comment:
👍 , looks good, was the recursive CTE just not terminating when there weren't any records with ROOT_ORDERED_TEST_PNTR = 1?

Reply:
Yep. It is better to run it if the condition exists.